submissions/andrewklmn/task_js_dom/app.js

```diff
@@ -0,0 +1,165 @@
+      location: "Venice, Santa Monica",
+      attractionMethod: "How to attract her: Don’t shave, use deodorant or have a job, and always carry around a little weed (Don’t worry, it’s practically legal here).",
+    },
+    {
+      menuTitle: 'Gold Digger',
+      title: "The Gold Digger",
+      image: "img/digger.jpg",
+      description: "Sadly, this is one of the more common types of L.A. women, but luckily they’re easy to spot. The twist here is that you don’t necessarily have to have a lot of money, you just have to have the potential to make it. So if you’re a junior agent or work in the mailroom at an entertainment law firm, she’ll invest her time in you. But if you don’t come through, buddy, you are screwed. All she wants is the house on the hill and the fancy car, so if you’re just looking for a trophy wife, then take your pick.",
+      location: "Beverly Hills, Hollywood Hills, any Hills, really.",
+      attractionMethod: "How to attract her: What are you, an idiot? Just flash your cash!",
+    },
+  ],
+  activeIndex: 0,
+};
+
+const menuContainer = document.querySelector('.menu-container');
+const menuUl = document.querySelector('.menu');
+const wraper = document.querySelector('.wraper');
+const header = document.querySelector('.header');
+const image = document.querySelector('.picture');
+const description = document.querySelector('.description');
+const locationArea = document.querySelector('.habitat');
+const attractionMethod = document.querySelector('.method');
+
+const getViewWidth = () => Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);
+const maxAutoHideMenuWidth = 820;
+
+const toggleMenu = (event)=>{
+  
+  event.target.classList.toggle("burger");
+  
+  menuContainer.classList.toggle("opened") ;
+  wraper.classList.toggle("wide");
+
+  event.stopPropagation();
+  return false;
+};
+
+const menuItemClickListener = (event)=>{
+
+  if (event.target.matches('.menu-item')) {
+    state.activeIndex = event.target.parentNode.id;
+    renderPage(state);
+  
+    if (getViewWidth() < maxAutoHideMenuWidth) {
+      document.querySelector('.btn-toogle-menu').click(); 
+    }
+    event.stopPropagation();
+  };
+
+  return false;
+};
+
+const renderMenu = (state)=>{
+  
+  menuUl.innerHTML = "";
+
+  state.items.forEach((item,index)=>{
+    const li = document.createElement("li");   
+    li.id = index;
+
+    const a = document.createElement("a");
+    a.innerHTML = item.menuTitle;
+    a.classList.add("menu-item");
+    a.href = "javascript:void(0)";
+    
+    if (index==state.activeIndex) {
+      a.classList.toggle("active");
+    };
+
+    li.appendChild(a);
+    menuUl.appendChild(li);
+  });
+};
+
+const showTextContent = ()=>{
+  document.body.classList.remove("text-white");
+}
+
+const renderArticle = (content)=> {
+
+  document.body.classList.add("text-white");
```

Comment:
Why do you need to add this class here? Is it an imitation of the loading data process?

Reply:
Yes. This is just a little animation for the case when the image loading takes a longer time. The text disappears before and appears after the image loading.

Reply:
Got it. Maybe it will be better to show/hide some loader instead of adding white color for the texts.